Source/Preprocessor/cpp.c

```diff
@@ -1280,11 +1280,23 @@ static DOH *Preprocessor_replace(DOH *s) {
       Replaceall(fn, "\\", "\\\\");
       Printf(ns, "\"%s\"", fn);
       Delete(fn);
-    } else if (Getattr(symbols, id)) {
-      DOH *e;
+    } else if ((m = Getattr(symbols, id))) {
       /* Yes.  There is a macro here */
+      /* If it expects arguments, they must come from `line_file` */
+      DOH *args = 0;
+      DOH *e;
+      int macro_additional_lines = 0;
       /* See if the macro expects arguments */
-      e = expand_macro(id, 0, s);
+      if (Getattr(m, kpp_args) && line_file) {
+        /* Yep.  We need to go find the arguments and do a substitution */
+        int line = Getline(line_file);
+        args = find_args(line_file, 1, id);
+        macro_additional_lines = Getline(line_file) - line;
+        assert(macro_additional_lines >= 0);
```

Comment:
I don't understand what this assertion is trying to achieve.

If you're wanting to assert that we found parameters, `assert(args);` would achieve that more directly (and probably more successfully - it looks to me like if we don't find parameters then `macro_additional_lines` will be zero anyway).

Reply:
I think it is somekind of leftover protection for a bug in the macro parsing - maybe something that was not clear if it was fixed or not. I got it from line 1212 and since the original author kept it, I decided to keep it too. It is not about `args` being non-zero which is, first, is a valid situation that is handled and, second, it cannot arise from calling `find_args`. Maybe remove both of them?